src/blob/index/simple.rs

```diff
@@ -278,6 +278,17 @@ impl Simple {
             0
         }
     }
+
+    fn mark_all_as_deleted_in_memory(headers: &mut InMemoryIndex, key: &[u8]) -> Option<u64> {
+        debug!("headers: {}", headers.len());
+        if let Some(vec_headers) = headers.get_mut(key) {
```

Comment:
I think it's better to return directly `headers.get_mut(key).map(|vec_headers| { ... })`